# Problem Statement 2:-->

A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.

File: LabTAT.csv

# Business Objectives-->

Ensure efficiency in report generation

compare laboratories performance

improve patient experience-->faster report means quicker treatment desicion

# Constraints-->

data normality assumption-->data must be normally distributed

varience assumption-->the variance across the four groups should be approximately equal

In [9]:
#import required libraries
import pandas as pd
import numpy as np
import scipy 
from scipy import stats

In [2]:
#load the dataset
df=pd.read_csv("lab_tat_updated.csv")

In [4]:
#print the 1st 5 rows of dataset
df.head(5)

,Laboratory_1,Laboratory_2,Laboratory_3,Laboratory_4
0,185.35,165.53,176.70,166.13
1,170.49,185.91,198.45,160.79
2,192.77,194.92,201.23,185.18
3,177.33,183.00,199.61,176.42
4,193.41,169.57,204.63,152.60


In [5]:
#print information of dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120 entries, 0 to 119
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Laboratory_1  120 non-null    float64
 1   Laboratory_2  120 non-null    float64
 2   Laboratory_3  120 non-null    float64
 3   Laboratory_4  120 non-null    float64
dtypes: float64(4)
memory usage: 3.9 KB


In [6]:
#print description of dataset
df.describe()

,Laboratory_1,Laboratory_2,Laboratory_3,Laboratory_4
count,120.000000,120.000000,120.000000,120.00000
mean,178.257333,178.902917,200.210167,163.68275
std,13.919668,14.957114,15.794801,15.08508
min,140.250000,140.550000,170.580000,124.06000
25%,170.267500,168.025000,190.182500,154.05000
50%,179.055000,178.870000,198.610000,164.42500
75%,187.222500,189.112500,211.197500,172.88250
max,216.390000,217.860000,238.700000,205.18000


In [12]:
#dataset columns
df.columns=["lab_1","lab_2","lab_3","lab_4"]
df.columns

Index(['lab_1', 'lab_2', 'lab_3', 'lab_4'], dtype='object')

In [13]:
#shape of dataset
df.shape

(120, 4)

Assumption of normality testing:

H0 (Null Hypothesis): Data follows a normal distribution.

H1 (Alternative Hypothesis): Data does not follow a normal distribution.

In [15]:
#Normality test
print("Lab 1 normality: ",stats.shapiro(df.lab_1))
print("Lab 2 normality: ",stats.shapiro(df.lab_2))
print("Lab 3 normality: ",stats.shapiro(df.lab_3))

Lab 1 normality:  ShapiroResult(statistic=0.9886695146560669, pvalue=0.4232051372528076)
Lab 2 normality:  ShapiroResult(statistic=0.9936320185661316, pvalue=0.8637352585792542)
Lab 3 normality:  ShapiroResult(statistic=0.9796065092086792, pvalue=0.06546738743782043)


Lab 1 p-value = 0.4232 > 0.05--> Fail to reject H0--> Data is normal 

Lab 2 p-value = 0.8637 > 0.05--> Fail to reject H0--> Data is normal 

Lab 3 p-value = 0.0655 > 0.05--> Fail to reject H0--> Data is normal 

Since all p-values are greater than 0.05, we fail to reject the null hypothesis.

The data follows a normal distribution for all laboratories.

In [16]:
#variance test
levene_test=scipy.stats.levene(df.lab_1,df.lab_2,df.lab_3)
print("Levene Test (Variance: ",levene_test)

Levene Test (Variance:  LeveneResult(statistic=1.5377583444327503, pvalue=0.21628196457198148)


H0-->Variances are equal across all laboratories

H1-->Variances are not equal across laboratories

p-value = 0.2163 > 0.05--> Fail to reject H0--> Variances are equal

In [17]:
#ANOVA test
anova_result = stats.f_oneway(df.lab_1, df.lab_2, df.lab_3) 
print("One_way ANOVA:", anova_result)

One_way ANOVA: F_onewayResult(statistic=84.23526440400498, pvalue=1.0799252415207089e-30)


Result p-value=1.0799

Interpretation:

H0: The mean Turn Around Time (TAT) is the same for all laboratories.

H1: At least one laboratory has a significantly different mean TAT.

Since p-value < 0.05, we reject the null hypothesis (H0).

At least one laboratory has a significantly different mean TAT compared to others.